File: src/main/java/fr/bfr/model/Character.java
package fr.bfr.model;

import lombok.*;

import javax.persistence.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Entity
@Table(name = "t_character")
public class Character implements Comparable<Character> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private Integer rarity;

    @Override
    public int compareTo(Character character) {
        if (this.rarity > character.rarity)
            return 1;
        else if (this.rarity.equals(character.rarity))
            return 0;
        else
            return -1;
    }
}

File: src/main/kotlin/fr/bfr/model/Character.kt
package fr.bfr.model

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "t_character")
class Character(
    @Id
    var id: Long,
    var name: String,
    var rarity: Int,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Character

        if (id != other.id) return false
        if (name != other.name) return false
        if (rarity != other.rarity) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + name.hashCode()
        result = 31 * result + rarity
        return result
    }
}
